### DISCUSSION TITLE
I can't use LM Studio in the new Fabric installed via GO, pls help

### DISCUSSION THREAD

**Discussion by poezie** [danielmiessler/Fabric#857] [Q&A]
I have move to the new Fabric install using Go , which has introduced a few more things. One query I have is I used to use LM Studio instead of Ollama , but atm it appears only Ollama is support . If that is not the case please can someone advise how I can setup the new fabric install and use my local Lm studio install instead of Ollama ?

**Comment by MrDowntempo** [comment]
I'm having the same issue. I tried to adapt the LM-Studio URLs to the ollama part of fabric -setup, but that of course did not work.

**Reply by juliomonteiro21** [reply]
I believe I could create a fork and change the open ai endpoint to the lm studio endpoint. During setup, register the lm-studio api-key.

Important: I'm not a programmer and I'm not sure if I would need any more changes than these. But if they don't make an update to make it possible to use lm studio, I'll try to do it that way...

**Reply by poezie** [reply]
I did play around with this but couldn't get it working , if you do get a working version pls do advise what you did ?

**Reply by mcox00941** [reply]
I'm having the same issue.  I'm trying various .env settings like we had with LMStudio and Fabric under Python, but I haven't figured it out yet. Do we have to have the OpenAI support like before or can we trick it into using LMStudio under the Ollama vendor model? 

```
Here are the settings I have in the setup wizard.

xxxxxxx:~$ fabric --setup


[Gemini]

Enter your Gemini API KEY (leave empty to skip):

[Gemini] skipped


[Anthropic]

Enter your Anthropic API KEY (leave empty to skip):

[Anthropic] skipped


[OpenAI]

Enter your OpenAI API KEY (leave empty to skip):
lmstudio
[OpenAI] configured


[Azure]

Enter your Azure API KEY (leave empty to skip):

[Azure] skipped


[Ollama]

Enter your Ollama URL (as a reminder, it is usually http://localhost:11434) (leave empty to skip):
http://192.168.12.175:8000/v1
[Ollama] configured


[Grocq]

Enter your Grocq API KEY (leave empty to skip):

[Grocq] skipped
OpenAI error, status code: 401, message: Incorrect API key provided: lmstudio. You can find your API key at https://platform.openai.com/account/api-keys.

Available vendor models:

Ollama



[Default]

Enter the index the name of your default model (leave empty to skip):

DEFAULT_MODEL=, is not valid
```

When I run fabric -L I get:

```
fabric can't initialize, please run the --setup procedure DEFAULT_VENDOR=, is not valid
DEFAULT_VENDOR=, is not valid
```

**Reply by poezie** [reply]
I also haven't managed to get it working and have tried multiple configuration options , I suspect it isn't supported currently , I am guessing

**Reply by poezie** [reply]
Has anyone been able to get Lm Studio to work with the new Fabric ?

**Reply by danhuss** [reply]
LM Studio mimics an OpenAI API endpoint which has a different structure from Ollama. It looks like the workaround to manually change the OPENAI_BASE_URL environment variable isn't working in the GO version. #883

**Comment by Fiddler2069** [comment]
How about just adding LM Studio support? He's got just about everything else. Hopefully we will get this sometime.
